internal/provider/resource_gitlab_group_saml_link.go

```diff
@@ -0,0 +1,163 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_group_saml_link", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_group_saml_link`" + ` resource allows to manage the lifecycle of an SAML integration with a group.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/groups.html#saml-group-links)`,
+
+		CreateContext: resourceGitlabGroupSamlLinkCreate,
+		ReadContext:   resourceGitlabGroupSamlLinkRead,
+		DeleteContext: resourceGitlabGroupSamlLinkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGitlabGroupSamlLinkImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: "The id of the GitLab group.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"access_level": {
+				Description:      fmt.Sprintf("Minimum access level for members of the SAML group. Valid values are: %s", renderValueListForDocs(validGroupSamlLinkAccessLevelNames)),
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validGroupSamlLinkAccessLevelNames, false)),
+				Required:         true,
+				ForceNew:         true,
+			},
+			"saml_group_name": {
+				Description: "The name of the SAML group.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"force": {
```

Comment:
This seems to be an anti-pattern to me - wouldn't the proper solution be to import existing SAML links into state?